tsl/src/continuous_aggs/common.h

```diff
@@ -0,0 +1,148 @@
+	int original_query_resno;
+	/*
+	 * "Original variables" are the Var nodes of the target list of the original
+	 * CREATE MATERIALIZED VIEW query. "Mapped variables" are the Var nodes of the materialization
+	 * table columns. The partialization query is the one that populates those columns. The
+	 * finalization query should use the "mapped variables" to populate the user view.
+	 */
+	List *orig_vars; /* List of Var nodes that have been mapped to materialization table columns */
+	List *mapped_vars; /* List of Var nodes of the corresponding materialization table columns */
+					   /* orig_vars and mapped_vars lists are mapped 1 to 1 */
+} AggPartCxt;
+
+/* Note that we set rowsecurity to false here. */
+#define CAGG_MAKEQUERY(selquery, srcquery)                                                         \
+	do                                                                                             \
+	{                                                                                              \
+		(selquery) = makeNode(Query);                                                              \
+		(selquery)->commandType = CMD_SELECT;                                                      \
+		(selquery)->querySource = (srcquery)->querySource;                                         \
+		(selquery)->queryId = (srcquery)->queryId;                                                 \
+		(selquery)->canSetTag = (srcquery)->canSetTag;                                             \
+		(selquery)->utilityStmt = copyObject((srcquery)->utilityStmt);                             \
+		(selquery)->resultRelation = 0;                                                            \
+		(selquery)->hasAggs = true;                                                                \
+		(selquery)->hasRowSecurity = false;                                                        \
+		(selquery)->rtable = NULL;                                                                 \
+	} while (0);
+
+#define CHUNKIDFROMRELID "chunk_id_from_relid"
+#define CONTINUOUS_AGG_CHUNK_ID_COL_NAME "chunk_id"
+
+#define MATPARTCOL_INTERVAL_FACTOR 10
+#define HT_DEFAULT_CHUNKFN "calculate_chunk_interval"
+#define CAGG_INVALIDATION_TRIGGER "continuous_agg_invalidation_trigger"
+#define BOUNDARY_FUNCTION "cagg_watermark"
+#define INTERNAL_TO_DATE_FUNCTION "to_date"
+#define INTERNAL_TO_TSTZ_FUNCTION "to_timestamp"
+#define INTERNAL_TO_TS_FUNCTION "to_timestamp_without_timezone"
+#define CONTINUOUS_AGG_MAX_JOIN_RELATIONS 2
+
+extern CAggTimebucketInfo cagg_validate_query(const Query *query, const bool finalized,
+											  const char *cagg_schema, const char *cagg_name);
+extern Query *destroy_union_query(Query *q);
+extern bool function_allowed_in_cagg_definition(Oid funcid);
+extern Var *mattablecolumninfo_addentry(MatTableColumnInfo *out, Node *input,
+										int original_query_resno, bool finalized,
+										bool *skip_adding);
+extern Oid relation_oid(NameData schema, NameData name);
```

Comment:
In the future we can generalize this one and move to utils.h and change all the places that do a combination of  `get_relname_relid`, but this can be material for a following PR.

Reply:
Sent a small PR to refactor it: https://github.com/timescale/timescaledb/pull/5716